src/test/java/de/tum/in/www1/artemis/AttachmentUnitIntegrationTest.java

```diff
@@ -124,7 +136,11 @@ public void updateAttachmentUnit_notInstructorInCourse_shouldReturnForbidden() t
     @Test
     @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
     public void updateAttachmentUnit_noId_shouldReturnBadRequest() throws Exception {
-        this.attachmentUnit = attachmentUnitRepository.save(attachmentUnit);
+        lecture1 = lectureRepository.findByIdWithStudentQuestionsAndLectureUnits(lecture1.getId()).get();
+        lecture1.addLectureUnit(this.attachmentUnit);
+        lectureRepository.save(lecture1);
+        this.attachmentUnit = (AttachmentUnit) lectureRepository.findByIdWithStudentQuestionsAndLectureUnits(lecture1.getId()).get().getLectureUnits().stream().findFirst().get();
+
```

Comment:
this code is duplicated 4 times. Try to create a function for that.

Reply:
done in e49e3394f2a7ebe0249b9c45bb625af3cd98c356